src/components/grid-list/grid-list.spec.ts

```diff
@@ -0,0 +1,384 @@
+    var template = `
+      <md-grid-list cols="1" rowHeight="100px">
+        <md-grid-tile></md-grid-tile>
+        <md-grid-tile></md-grid-tile>
+      </md-grid-list>
+    `;
+
+    builder.overrideTemplate(TestGridList, template)
+        .createAsync(TestGridList).then((fixture: ComponentFixture<TestGridList>) => {
+          fixture.detectChanges();
+          let list = fixture.debugElement.query(By.directive(MdGridList));
+          expect(getProp(list, 'height')).toBe('201px');
+        });
+  });
+
+  it('should allow adjustment of tile colspan', () => {
+    var template = `
+      <div style="width:400px">
+        <md-grid-list cols="4">
+          <md-grid-tile [colspan]="colspan"></md-grid-tile>
+        </md-grid-list>
+      </div>
+    `;
+
+    builder.overrideTemplate(TestGridList, template)
+        .createAsync(TestGridList).then((fixture: ComponentFixture<TestGridList>) => {
+          fixture.componentInstance.colspan = 2;
+          fixture.detectChanges();
+
+          let tile = fixture.debugElement.query(By.directive(MdGridTile));
+          expect(getProp(tile, 'width')).toBe('199.5px');
+
+          fixture.componentInstance.colspan = 3;
+          fixture.detectChanges();
+          expect(getProp(tile, 'width')).toBe('299.75px');
+        });
+  });
+
+  it('should allow adjustment of tile rowspan', () => {
+    var template = `
+      <md-grid-list cols="1" rowHeight="100px">
+        <md-grid-tile [rowspan]="rowspan"></md-grid-tile>
+      </md-grid-list>
+    `;
+
+    builder.overrideTemplate(TestGridList, template)
+        .createAsync(TestGridList).then((fixture: ComponentFixture<TestGridList>) => {
+          fixture.componentInstance.rowspan = 2;
+          fixture.detectChanges();
+
+          let tile = fixture.debugElement.query(By.directive(MdGridTile));
+          expect(getProp(tile, 'height')).toBe('201px');
+
+          fixture.componentInstance.rowspan = 3;
+          fixture.detectChanges();
+          expect(getProp(tile, 'height')).toBe('302px');
+        });
+  });
+
+  it('should lay out tiles correctly for a complex layout', () => {
+    var template = `
+      <div style="width:400px">
+        <md-grid-list cols="4" rowHeight="100px">
+          <md-grid-tile *ngFor="let tile of tiles" [colspan]="tile.cols" [rowspan]="tile.rows"
+                        [style.background]="tile.color">
+            {{tile.text}}
+          </md-grid-tile>
+        </md-grid-list>
+      </div>
+    `;
+
+    builder.overrideTemplate(TestGridList, template)
+        .createAsync(TestGridList).then((fixture: ComponentFixture<TestGridList>) => {
+          fixture.componentInstance.tiles = [
+            {text: 'One', cols: 3, rows: 1, color: 'lightblue'},
+            {text: 'Two', cols: 1, rows: 2, color: 'lightgreen'},
+            {text: 'Three', cols: 1, rows: 1, color: 'lightpink'},
+            {text: 'Four', cols: 2, rows: 1, color: '#DDBDF1'},
```

Comment:
Also these colors are boring. You should try some better ones like `sienna`, `mediumaquamarine`, or `lemonchiffon`.